iOS/Flows/Conversation/Conversation Header/MembersViewController.swift

```diff
@@ -39,28 +42,34 @@ class MembersViewController: DiffableCollectionViewController<MembersCollectionV
         ConversationsManager.shared.$activeConversation
             .removeDuplicates()
             .mainSink { [unowned self] conversation in
-                Task {
-                    guard let cid = conversation?.cid else {
-                        await self.dataSource.deleteAllItems()
+                self.loadDataTask?.cancel()
+
+                if let cid = conversation?.cid {
+                    self.conversationController = ConversationController.controller(cid)
+                } else {
+                    self.conversationController = nil
+                }
+
+                self.loadDataTask = Task { [weak self] in
+                    guard let conversationController = self?.conversationController else {
+                        // If there's no current conversation, then there's nothing to show.
+                        await self?.dataSource.deleteAllItems()
                         return
                     }
 
-                    let conversationController = ChatClient.shared.channelController(for: cid)
-                    self.conversationController = conversationController
-
-                    await self.loadData()
+                    await self?.loadData()
 
                     guard !Task.isCancelled else { return }
 
-                    self.subscribeToUpdates(for: conversationController)
-                }.add(to: self.autocancelTaskPool)
```

Comment:
I wanted to be able to cancel this Task specifically so I assigned it to its own variable.